### PR TITLE
Fix iop not being deleted during uninstall when specifying filename

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
Default installation and revisioned installation are fine, but if I have a custom IOP file, it will not be deleted. This currently may lead to other commands that depends on the IOP not working as expected.

### CODE REVIEW

**Comment by hanxiaop** [comment]
After this fix, iop can be deleted:
```
 istioctl uninstall -f out/yamls/overlay.yaml --revision=default
  ...
  Removed IstioOperator:istio-system:installed-state-istio-controlplane.
✔ Uninstall complete                             
```

**Comment by bleggett** [comment]
This is incompatible with https://github.com/istio/istio/pull/47860, right?

File: operator/cmd/mesh/uninstall.go [hunk 1]
```diff
@@ -200,6 +200,21 @@ func uninstall(cmd *cobra.Command, ctx cli.Context, rootArgs *RootArgs, uiArgs *
 	if err := h.DeleteControlPlaneByManifests(manifestMap, iop.Spec.Revision, uiArgs.purge); err != nil {
 		return fmt.Errorf("failed to delete control plane by manifests: %v", err)
 	}
+	iopName := savedIOPName(iop)
```
Comment (bleggett) [inline operator/cmd/mesh/uninstall.go:203]:
```	
iopName := savedIOPName(iop)
```

Relying on this is incompatible with https://github.com/istio/istio/pull/47860, I think?

Comment (hanxiaop) [reply operator/cmd/mesh/uninstall.go:203]: Yes this is incompatible. I wasn't excepting #47860 when I was trying to fix the current problem, but that is much desired solution. I'm going to close this one.

**Pull request closed without merging**